packages/mux-player/src/errors.ts

```diff
@@ -1,166 +1,182 @@
+  if (mediaError.code) {
+    if (mediaError.code === MediaError.MEDIA_ERR_NETWORK) return i18n(`Network Error`, translate);
+    if (mediaError.code === MediaError.MEDIA_ERR_DECODE) return i18n(`Media Error`, translate);
+    if (mediaError.code === MediaError.MEDIA_ERR_SRC_NOT_SUPPORTED) return i18n(`Source Not Supported`, translate);
+  }
 
-          if (!playbackToken) break;
+  return i18n(`Error`, translate);
+};
 
-          const { exp: tokenExpiry, aud: tokenType, sub: tokenPlaybackId } = parseJwt(playbackToken);
-          const tokenExpired = Date.now() > tokenExpiry * 1000;
-          const playbackIdMismatch = tokenPlaybackId !== playbackId;
-          const badTokenType = tokenType !== 'v';
-          const dateOptions: any = {
-            timeStyle: 'medium',
-            dateStyle: 'medium',
-          };
+const muxMediaErrorToDialogMessage = (mediaError: MediaError, translate = false) => {
+  if (mediaError.muxCode) {
+    const category = capitalizeFirstLetter(mediaError.errorCategory ?? 'video');
+    const tokenNamePrefix = errorCategoryToTokenNameOrPrefix(mediaError.errorCategory ?? MuxErrorCategory.VIDEO);
+    if (mediaError.muxCode === MuxErrorCode.NETWORK_OFFLINE) {
+      return i18n(`Check your internet connection and try reloading this video.`, translate);
+    }
+    if (mediaError.muxCode === MuxErrorCode.NETWORK_TOKEN_EXPIRED) {
+      return i18n(`The video’s secured {tokenNamePrefix}-token has expired.`, translate).format({
+        tokenNamePrefix,
+      });
+    }
+    if (mediaError.muxCode === MuxErrorCode.NETWORK_TOKEN_SUB_MISMATCH) {
+      return i18n(
+        `The video’s playback ID does not match the one encoded in the {tokenNamePrefix}-token.`,
+        translate
+      ).format({
+        tokenNamePrefix,
+      });
+    }
 
-          if (tokenExpired) {
-            dialog.title = i18n(`Video URL has expired`, translate);
-            dialog.message = i18n(`The video’s secured playback-token has expired.`, translate);
-            devlog.message = i18n(`The video’s secured playback-token has expired.`, translate);
-            devlog.context = i18n(`Expired at: {expiredDate}. Current time: {currentDate}.`, translate).format({
-              expiredDate: new Intl.DateTimeFormat(lang.code, dateOptions).format(tokenExpiry * 1000),
-              currentDate: new Intl.DateTimeFormat(lang.code, dateOptions).format(Date.now()),
-            });
-            devlog.file = '403-expired-token.md';
-            break;
-          }
+    if (mediaError.muxCode === MuxErrorCode.NETWORK_TOKEN_MALFORMED) {
+      return i18n(`{category} URL is formatted incorrectly`, translate).format({ category });
+    }
+    if (
+      [
+        MuxErrorCode.NETWORK_TOKEN_AUD_MISMATCH,
+        MuxErrorCode.NETWORK_TOKEN_AUD_MISSING,
+        // @ts-ignore
+      ].includes(mediaError.muxCode)
+    ) {
+      return i18n(`The {tokenNamePrefix}-token is formatted with incorrect information.`, translate).format({
```

Comment:
Nit: wonder if more specificity would help make *how* to fix whatever is wrong with the token easier. In this case calling out that it's the `aud` claim at issue.

Likewise for the other cases where something more specific can be inferred from the `MuxErrorCode`.

Reply:
Also assuming this is user-facing (please correct me if I am wrong) given `muxMediaErrorToDialogMessage`

Reply:
Yeah this is user-facing. Our assumption is the user will not be the developer (at least in the vast majority of cases). We've aimed for treating users as different "audiences" than devs with language appropriate to each. In the case of devs, they will have the errors themselves, the dev logs, and Mux Data. In the case of users, they'll have the error dialog UI. The goal (perhaps sometimes falling short) is to make each presentation of an error as actionable as possible for each audience. For the devs, that typically means describing as best as possible "root cause." For the end user, that will very much depend on what kind of actions they can('t) take. In this particular case, a more descriptive error message would be something like:

> It looks like there might be some bad code for loading playback: The {tokenNamePrefix}-token is formatted with incorrect information. You may need to raise an issue with the relevant folks to resolve.

Reply:
Right, that makes sense.